src/karts/controller/skidding_ai.cpp

```diff
@@ -1575,14 +1575,20 @@ void SkiddingAI::handleCake(int item_skill)
     }
 
     // Don't fire on an invulnerable kart.
-    if (m_kart_behind->isInvulnerable())
+    if (m_kart_behind != NULL)
     {
-        fire_behind -= 100.0f;
+        if (m_kart_behind->isInvulnerable())
+        {
+            fire_behind -= 100.0f;
+        }
```

Comment:
Don't nest those statements, use one if and &&. Also no need to use != NULL:
```
if(m_kart_behind && m_kart_behind->isInvulnerable())
```

Reply:
Ok. I wasn't sure if this was enough to avoid a segfault.

Reply:
Yes, C++ guarantees that those expressions are evaluated in order